mmdet/apis/train.py

```diff
@@ -140,8 +160,16 @@ def train_detector(model,
         eval_hook = DistEvalHook if distributed else EvalHook
         runner.register_hook(eval_hook(val_dataloader, **eval_cfg))
 
+        if nncf_enable_compression:
```

Comment:
I thought that you create EvalBeforeRunHook and DistEvalBeforeRunHook with the only before_run overrided method and make selection at line 160 depending on whether nncf enabled or not. That's it.

Reply:
Fixed